### PR TITLE
ASUIS-222: Display User Courses.

### PR DESCRIPTION
**Pull request by georgebaev**
@mlsamuelson this displays the User Courses.

Ref.: https://asudev.jira.com/browse/ASUIS-222

### CODE REVIEW

**Comment by georgebaev** [comment]
@duarte-daniela I restored the `request()` method at https://github.com/ASUWebPlatforms/asu-isearch/pull/99/files#diff-e6e6306e0ed62a89c3bf5e53682aa2d60f9ed141027e601d9ba025b77189e6a8R340 in order to be able to retrieve the Courses information based on the ASURITE ID value. 

Could you please advice if we have such Data Provider method which I could use in order to retrieve the User Profile courses? Thank you.

**Comment by duarte-daniela** [comment]
@georgebaev it's ok that you restored the method. The main function of the data provider is the connection with elastic, and the courses is a third party, so there is no method available for this purpose.

File: web/modules/custom/webdir_profile/webdir_profile.module [hunk 1]
```diff
@@ -348,3 +350,109 @@ function webdir_profile_preprocess_horizontal_tabs(&$variables) {
...
+    switch ($strm[0]) {
+      case '2':
+        $termCode .= '20';
+        break;
+      case '1':
+        $termCode .= '19';
+        break;
+
+      default:
+        // Don't know what to do with other values. Back to the future? :)
```
Comment (mlsamuelson) [inline web/modules/custom/webdir_profile/webdir_profile.module:403]: I hope by the future ASU has a better means of managing term codes. :D

Comment (georgebaev) [reply web/modules/custom/webdir_profile/webdir_profile.module:403]: I added a cutom Twig template at web/themes/custom/webdir/templates/field/field--user--field-profile-presentation.html.twig which displays the Courses static information just above the Presentations data.

File: web/modules/custom/webdir_profile/webdir_profile.module [hunk 2]
```diff
@@ -348,3 +350,109 @@ function webdir_profile_preprocess_horizontal_tabs(&$variables) {
 
   $variables['display_expertise'] = $display_expertise;
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function webdir_profile_preprocess_field__user__field_profile_presentation(&$variables) {
```
[the remark this answers is not in the document]
Comment (georgebaev) [reply web/modules/custom/webdir_profile/webdir_profile.module:357]: I added a cutom Twig template at web/themes/custom/webdir/templates/field/field--user--field-profile-presentation.html.twig which displays the Courses static information just above the Presentations data.

**Comment by georgebaev** [comment]
@mlsamuelson The courses are visible now at https://pr-99-asu-isearch.ws.asu.edu/user/59 but only after I added the `test` string in the Presentations. If no Presentation data exists, the Courses are not displayed either...

May i add a `Courses` field in the User profile. It could be a Boolean field and will act as a placeholder for the Courses data. What do you think?

**Comment by mlsamuelson** [comment]
@georgebaev Adding a courses field to the profile sounds like a great solution here. Please do that.

**Comment by georgebaev** [comment]
@mlsamuelson I added the `Courses` field and the Courses are displayed now: https://pr-99-asu-isearch.ws.asu.edu/user/59 . 

Please note that the Courses field is visible in the User Profile form. The field is accessible by the administrators only. We have to keep this field in the user form so that once we save the existing user's data, the related Courses value (0 or 1) is stored in the database. Otherwise, if no Courses field data is present in the database, the field's Twig template is not processed and no Courses are displayed.

**Review by mlsamuelson — APPROVED** [review]
Thank you for resolving the issue.

**Pull request merged**